nilearn/conftest.py

```diff
@@ -112,6 +112,23 @@ def warnings_as_errors():
         yield
 
 
+@pytest.fixture(autouse=True)
+def suppress_specific_warning():
+    """Ignore internal deprecation warnings."""
+    with warnings.catch_warnings():
+        messages = (
+            "The `darkness` parameter will be deprecated.*|"
+            "`legacy_format` will default to `False`.*|"
+            "In release 0.13, this fetcher will return a dictionary.*|"
```

Comment:
Thanks for finishing the PR @Remi-Gau !
Just fyi I did not check that this was a thorough list of internal deprecation messages.

Reply:
will quickly check but your work should allow to make it easy to suppress if there are more of those warnings